codegen/src/shuttle_main/mod.rs

```diff
@@ -14,9 +14,13 @@ pub(crate) fn r#impl(_attr: TokenStream, item: TokenStream) -> TokenStream {
     let loader = Loader::from_item_fn(&mut fn_decl);
 
     quote! {
-        #[tokio::main]
-        async fn main() {
-            shuttle_runtime::start(loader).await;
+        fn main() {
+            use ::shuttle_runtime::tokio;
+            #[tokio::main]
+            async fn main() {
```

Comment:
Why is there a `main()` inside `main`?

Reply:
To scope off our tokio import and not leak it to the end user's main.rs

Reply:
Maybe we should not call the `tokio::main` macro then and just expand it ourselves.
https://docs.rs/tokio/latest/tokio/attr.main.html#using-the-multi-thread-runtime

Reply:
I'm trying to reduce the confusion here for the future